### PR TITLE
GitHub Issue NOAA-EMC/GSI#24. Add LeoGeo AMVs in GSI.

### PR DESCRIPTION
**Pull request by ilianagenkova**
AMV_Genkova_parallel_ncio_LG - branch adding LeoGeo AMVs to GSI

Please see the GitHub Issue [#24](https://github.com/NOAA-EMC/GSI/issues/24) for more details.

### CODE REVIEW

**Comment by XiujuanSu-NOAA** [comment]
Hi Mike

I looked at the wiki page on github and I cannot find instructions how to
review the ticket, should I go to vlab?  Thanks

Su

**Comment by MichaelLueken-NOAA** [comment]
Hi Su,

We no longer use VLab, only GitHub.  To review a pull request, click on Files changed.  The changes are now shown.  If you see something that you have questions about, click the blue + icon and add a comment.

For more details, please see https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews

Mike

**Comment by XiujuanSu-NOAA** [comment]
I checked global_convinfo.txt file, the changes are fine.  However should the changes also apply to other convinfo files?

Su

File: src/gsi/read_satwnd.f90 [hunk 1]
```diff
@@ -402,6 +402,12 @@ subroutine read_satwnd(nread,ndata,nodata,infile,obstype,lunout,gstime,twind,sis
                  itype=240
               endif
            endif
+        else if( trim(subset) == 'NC005072') then                   ! LEOGEO (LeoGeo) winds
+           if(hdrdat(1) == 854 ) then                               ! LeoGeo satellite ID
+              if(hdrdat(9) == one)  then                            ! LEOGEO IRwinds
+                 itype=255
```
[the remark this answers is not in the document]
Comment (ilianagenkova) [reply src/gsi/read_satwnd.f90:408]: Thank you, Su! I have the two "if" checks on separate lines for better readability and for consistency with the other winds types.

**Comment by ilianagenkova** [comment]
Su,
we decided not to make changes to the historic convinfo files, because LeoGeo winds do not exist before Sep 2019.  We only changed the global_convinfo.txt because these winds will be in monitoring mode for a while. When we are ready to start assimilating them, we will consider changing the other convinfo files. Thanks!

**Comment by XiujuanSu-NOAA** [comment]
Ok, no problem.

Su

File: src/gsi/read_satwnd.f90 [hunk 2]
```diff
@@ -868,6 +874,33 @@ subroutine read_satwnd(nread,ndata,nodata,infile,obstype,lunout,gstime,twind,sis
...
+                    !if(qcdat(2,j) <= r10000 .and. qcdat(3,j) <r10000 ) then
+                    !   if(qcdat(2,j) ==  one  .and. qifn >r105) then
+                    !      qifn=qcdat(3,j)
+                    !   else if(qcdat(2,j) ==  three .and. qify >r105) then
+                    !      qify=qcdat(3,j)
+                    !   else if( qcdat(2,j) == four .and. ee >r105) then
+                    !      ee=qcdat(3,j)
+                    !   endif
+                    !endif
+                    !enddo
```
Comment (MichaelLueken-NOAA) [inline src/gsi/read_satwnd.f90:902]: If these commented out lines aren't needed and won't be used in the future, please remove them.

**Comment by ilianagenkova** [comment]
Hi Mike,

These lines will be used in the near future. NESDIS is working towards unifying the BUFR formats of all their winds and by the end of the year I'd know the final configuration. I could then clean up the code from this and other comments which might become obsolete. Thanks!

**Comment by MichaelLueken-NOAA** [comment]
The due date for the review committee has passed with no feedback, so final approval was given and the update was merged to the NOAA-EMC/GSI master.

**Pull request merged**